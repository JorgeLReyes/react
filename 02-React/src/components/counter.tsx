import { useState } from "react";
export const Counter = () => {
  const [counter, setCounter] = useState<number>(0);

  // e: React.MouseEvent<HTMLButtonElement>
  const handleCounter = (number: number) =>
    setCounter((counter) => counter + number);

  const handleReset = () => setCounter(0);

  return (
    <>
      <h1>CounterApp</h1>
      <p>{counter}</p>
      <button onClick={() => handleCounter(-1)}>-1</button>
      <button onClick={handleReset}>0</button>
      <button onClick={() => handleCounter(+1)}>+1</button>
    </>
  );
};
