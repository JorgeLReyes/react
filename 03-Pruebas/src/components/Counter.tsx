import { useState } from "react";
export const Counter = ({ value = 0 }: { value?: number }) => {
  const [counter, setCounter] = useState<number>(value);

  // e: React.MouseEvent<HTMLButtonElement>
  const handleCounter = (number: number) =>
    setCounter((counter) => counter + number);

  const handleReset = () => setCounter(0);

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={() => handleCounter(-1)}>-1</button>
      <button aria-label="btn-reset" onClick={handleReset}>
        0
      </button>
      <button onClick={() => handleCounter(+1)}>+1</button>
    </>
  );
};
