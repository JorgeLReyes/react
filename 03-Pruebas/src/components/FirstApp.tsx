import PropTypes from "prop-types";

interface Props {
  title: string;
  subTitle: string;
  name: string;
}

const FirstApp = ({ title, subTitle, name }: Props) => {
  return (
    <div>
      <h1 data-testid="test-title">{title}</h1>
      <h2>{subTitle}</h2>
      <h2>{subTitle}</h2>
      <p>Hola, {name}!</p>
    </div>
  );
};

// FirstApp.defaultProps = {
//   title: "Hello World",
//   subTitle: "Welcome to my app",
//   // name: "Ingrid",
// };

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default FirstApp;
