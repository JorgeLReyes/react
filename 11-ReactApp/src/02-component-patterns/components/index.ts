export * from "./ProductButtons";
export * from "./ProductImage";
export * from "./ProductTitle";
// export * from "./ProductCard";

import { ProductCard as ProductCardHOC } from "./ProductCard";
import { ProductTitle } from "./ProductTitle";
import { ProductImage } from "./ProductImage";
import { ProductButtons } from "./ProductButtons";
import { ProductCardComponents } from "../interfaces/interfaces";

export const ProductCard: ProductCardComponents = Object.assign(
  ProductCardHOC,
  {
    Title: ProductTitle,
    Image: ProductImage,
    Buttons: ProductButtons,
  }
);
