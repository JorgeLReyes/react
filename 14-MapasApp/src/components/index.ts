export * from "./MapView";
export * from "./Loading";
export * from "./BtnLocation";
export * from "./ReactLogo";
export * from "./SearchBar";
export * from "./SearchResult";
