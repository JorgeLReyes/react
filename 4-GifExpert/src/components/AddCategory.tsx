import { useState } from "react";

interface Props {
  onAddCategory: (category: string) => boolean;
}

export const AddCategory = ({ onAddCategory }: Props) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = ({
    target,
  }: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(target.value);
  };

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (!inputValue) return;
    if (onAddCategory(inputValue)) {
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={handleInputChange}
      />
      <button type="submit">Enviar</button>
    </form>
  );
};
