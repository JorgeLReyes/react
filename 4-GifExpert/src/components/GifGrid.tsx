import { GifItem, Loader } from "./";
import useFetchGifs from "../hooks/useFetchGifs";

interface Props {
  name: string;
}

export const Category = ({ name }: Props) => {
  const { images, isLoading } = useFetchGifs(name);
  return (
    <section>
      <h2>{name}</h2>
      <section className="card-grid">
        <Loader isRender={isLoading} />
        {images.map((img) => (
          <GifItem key={img.id} {...img} />
        ))}
      </section>
    </section>
  );
};
