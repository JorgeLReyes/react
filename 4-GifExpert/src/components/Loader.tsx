import "./Loader.css";
export const Loader = ({ isRender }: { isRender: boolean }) => {
  if (!isRender) return null;
  return (
    <div className="three-body">
      <div className="three-body__dot"></div>
      <div className="three-body__dot"></div>
      <div className="three-body__dot"></div>
    </div>
  );
};
