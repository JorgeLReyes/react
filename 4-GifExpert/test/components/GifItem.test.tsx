import { describe, test, expect } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";
import React from "react";
describe("Gif Item Component", () => {
  const props = {
    title: "Test Gif",
    url: "https://test.com/test.gif",
  };

  test("should be have properties title and url", () => {
    const { container } = render(<GifItem {...props} />);

    const img = container.querySelector(`img[src="${props.url}"]`);
    expect(img).toBeTruthy();
    expect(img?.getAttribute("src")).toBe(props.url);
    expect(img?.getAttribute("alt")).toBe(props.title);

    expect(screen.getByRole("img").getAttribute("src")).toBe(props.url);
    expect(screen.getByRole("img").getAttribute("alt")).toBe(props.title);
    // expect(screen.getByRole("img").alt).toBe(props.title);

    expect(screen.getByText(props.title)).toBeTruthy();
  });

  test("should be match to screen", () => {
    const { container } = render(<GifItem {...props} />);
    expect(container).toMatchSnapshot();
  });
});
