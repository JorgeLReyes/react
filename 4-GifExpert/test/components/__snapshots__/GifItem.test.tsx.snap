// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Item Component should be match to screen 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test Gif"
      src="https://test.com/test.gif"
    />
    <p>
      Test Gif
    </p>
  </div>
</div>
`;
